kahuna/public/js/services/api/collections-api.js

```diff
@@ -13,13 +13,15 @@ collectionsApi.factory('collections',
                        ['$rootScope', '$q', 'mediaApi', 'imageAccessor', 'apiPoll',
                         function ($rootScope, $q, mediaApi, imageAccessor, apiPoll) {
 
+    const collectionsRoot = mediaApi.root.follow('collections');
+
     // TODO: Rx?
     let collections;
 
     function getCollections() {
+        // TODO: do we want to memoize? what if we want to reload?
         if (! collections) {
-            collections = mediaApi.root.follow('collections').get().
-                then(collectionsService => collectionsService.follow('collections').get());
```

Comment:
Main thing to note is that you can do `mediaApi.root.follow('collections').follow('collections').get()` since resources are lazy that way.